Mathlib/GroupTheory/GroupAction/MultipleTransitivity.lean

```diff
@@ -298,8 +299,195 @@ end Higher
 
 end MulAction
 
+namespace SubMulAction.ofStabilizer
+
+variable {G α : Type*} [Group G] [MulAction G α]
+
+open scoped BigOperators Pointwise Cardinal
+
+@[to_additive]
+theorem isMultiplyPretransitive_iff_of_conj
+    {n : ℕ} {a b : α} {g : G} (hg : b = g • a) :
+    IsMultiplyPretransitive (stabilizer G a) (ofStabilizer G a) n ↔
+      IsMultiplyPretransitive (stabilizer G b) (ofStabilizer G b) n :=
+  IsPretransitive.of_embedding_congr (MulEquiv.surjective _) (ofStabilizer.conjMap_bijective hg)
+
+@[to_additive]
+theorem isMultiplyPretransitive_iff [IsPretransitive G α] {n : ℕ} {a b : α} :
+    IsMultiplyPretransitive (stabilizer G a) (ofStabilizer G a) n ↔
+      IsMultiplyPretransitive (stabilizer G b) (ofStabilizer G b) n :=
+  let ⟨_, hg⟩ := exists_smul_eq G a b
+  isMultiplyPretransitive_iff_of_conj hg.symm
+
+/-- Multiple transitivity of a pretransitive action
+is equivalent to one less transitivity of stabilizer of a point
+(Wielandt, th. 9.1, 1st part) -/
+@[to_additive /-- Multiple transitivity of a pretransitive action
+is equivalent to one less transitivity of stabilizer of a point
+[Wielandt, th. 9.1, 1st part][Wielandt-1964]. -/]
+theorem isMultiplyPretransitive [IsPretransitive G α] {n : ℕ} {a : α} :
+    IsMultiplyPretransitive G α n.succ ↔
+      IsMultiplyPretransitive (stabilizer G a) (SubMulAction.ofStabilizer G a) n := by
+  refine ⟨fun hn ↦ ⟨fun x y ↦ ?_⟩, fun hn ↦ ⟨fun x y ↦ ?_⟩⟩
+  · obtain ⟨g, hgxy⟩ := exists_smul_eq G (ofStabilizer.snoc x) (ofStabilizer.snoc y)
+    have hg : g ∈ stabilizer G a := by
+      rw [mem_stabilizer_iff]
+      rw [DFunLike.ext_iff] at hgxy
+      convert hgxy (last n) <;> simp [smul_apply, ofStabilizer.snoc_last]
+    use ⟨g, hg⟩
+    ext i
+    simp only [smul_apply, SubMulAction.val_smul_of_tower, subgroup_smul_def]
+    rw [← ofStabilizer.snoc_castSucc x, ← smul_apply, hgxy, ofStabilizer.snoc_castSucc]
+  · -- gx • x = x1 :: a
+    obtain ⟨gx, x1, hgx⟩ := exists_smul_of_last_eq G a x
+    -- gy • y = y1 :: a
+    obtain ⟨gy, y1, hgy⟩ := exists_smul_of_last_eq G a y
+    -- g • x1 = y1,
+    obtain ⟨g, hg⟩ := hn.exists_smul_eq x1 y1
+    use gy⁻¹ * g * gx
+    ext i
+    simp only [mul_smul, smul_apply, inv_smul_eq_iff]
+    simp only [← smul_apply _ _ i, hgy, hgx]
+    simp only [smul_apply]
+    rcases Fin.eq_castSucc_or_eq_last i with ⟨i, rfl⟩ | ⟨rfl⟩
+    · simp [ofStabilizer.snoc_castSucc, ← hg, SetLike.val_smul, subgroup_smul_def]
+    · simp only [ofStabilizer.snoc_last, ← hg]
+      exact g.prop
+
+end ofStabilizer
+
+namespace ofFixingSubgroup
+
+variable {G α : Type*} [Group G] [MulAction G α]
+
+open SubMulAction Fin.Embedding
+
+variable (G) in
+/-- The fixator of a finite subset of cardinal d in an n-transitive action
```

Comment:
I just noticed this: here `fixator` and `fixingSubgroup` are the same, right?

If you have a preference for using a different name, that's fine, but otherwise, would you mind aligning the names?